test/manifold_test.cpp

```diff
@@ -215,6 +216,15 @@ TEST(Manifold, Revolve2) {
   EXPECT_NEAR(prop.surfaceArea, 96.0f * glm::pi<float>(), 1.0f);
 }
 
+TEST(Manifold, PartialRevolve) {
+  Polygons polys = SquareHole(2.0f);
+  Manifold donutHole = Manifold::Revolve(polys, 48, 180);
+  EXPECT_EQ(donutHole.Genus(), 1);
```

Comment:
Would you mind pasting an image of this object? I'm having trouble visualizing what axis it's rotating around.

Reply:
Yes, it looks like this. I'd definitely want to do more thorough testing before merging (even if we don't want to include them all).
![PartialRevolve](https://github.com/elalish/manifold/assets/26554374/8efa9aca-223b-40af-8691-187e485e036d)

Reply:
Thanks; I'd recommend pushing it either in or out a little, since right now it's just rounding error determining a major topological change in the center.